File: Java/Ejercicio 1/ModoEvaluacion.java
import java.util.Collections;
import java.util.Comparator;
import java.util.Scanner;

public class ModoEvaluacion {
    public static void iniciarEvaluacion() {
        if (GestionVocabulario.vocabulario.isEmpty()) {
            System.out.println("No hay palabras.");
            return;
        }

        // Ordenar por errores
        Collections.sort(GestionVocabulario.vocabulario,
                Comparator.comparingInt((Palabra p) -> p.errores).reversed());

        Scanner sc = new Scanner(System.in);
        for (Palabra p : GestionVocabulario.vocabulario.toArray(new Palabra[0])) {
            System.out.print("Traduce al español: " + p.palabra + " → ");
            String resp = sc.nextLine().toLowerCase();

            if (resp.equals(p.significado)) {
                p.aciertos++;
                System.out.println("Correcto!");
                if (p.aciertos >= 5) {
                    System.out.println("Palabra aprendida: " + p.palabra);
                    GestionVocabulario.vocabulario.remove(p);
                }
            } else {
                System.out.println("Incorrecto. Respuesta Correcta: " + p.significado);
                p.aciertos = 0;
                p.errores++;
            }
        }
    }
}

File: Typescript/Ejercicio 1/modoEvaluacion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.iniciarEvaluacion = iniciarEvaluacion;
var gestionVocabulario_1 = require("./gestionVocabulario");
var readline_sync_1 = require("readline-sync");
function iniciarEvaluacion() {
    if (gestionVocabulario_1.vocabulario.length === 0) {
        console.log("No hay palabras.");
        return;
    }
    gestionVocabulario_1.vocabulario.sort(function (a, b) { return b.errores - a.errores; });
    for (var i = 0; i < gestionVocabulario_1.vocabulario.length; i++) {
        var item = gestionVocabulario_1.vocabulario[i];
        var resp = (0, readline_sync_1.question)("Traduce al espa\u00F1ol: \"".concat(item.palabra, "\" -> ")).toLowerCase();
        if (resp === item.significado) {
            item.aciertos++;
            console.log("Correcto!");
            if (item.aciertos >= 5) {
                console.log("Aprendida: ".concat(item.palabra));
                gestionVocabulario_1.vocabulario.splice(i, 1);
                i--;
            }
        }
        else {
            console.log("Incorrecto. Respuesta: ".concat(item.significado));
            item.aciertos = 0;
            item.errores++;
        }
    }
}
